tasks/javascript.js

```diff
@@ -105,18 +113,25 @@ gulp.task('javascript:lint', function () {
 
 // karma unit tests in local library only
 gulp.task('javascript:unit', function () {
-  return streams.jsSpec()
-    .pipe(browserify
-      .compile(80, transforms.concat(browserify.jasmineTransform('@')))
-      .all('index.js'))
+  return combined.create()
+    .append(
+      streams
+        .testDependencies({
+          dev: true,
+          read: false
+        })
+    )
+    .append(
+      streams
+        .jsSpec()
+        .pipe(browserify
+          .compile(80, transforms.concat(browserify.jasmineTransform('@')))
+          .all('index.js'))
+        .pipe(gulpIf('*.js', gulp.dest(streams.TEST)))
```

Comment:
Ensure all files go to `gulp.dest()` but filter the output thereafter to include just the `index.js`.
i.e. use gulp-filter instead of gulp-if